sunpy/net/tests/test_scraper.py

```diff
@@ -225,3 +225,27 @@ def test_yearly_overlap():
     # Should return a single file for 2013
     trange = TimeRange("2013-01-02", "2013-01-03")
     assert len(scraper.filelist(trange)) == 1
+
+@pytest.fixture(params=[(400,1) , (403,1) , (429,6) , (504,6)])
+def endpoint(request):
+    return request.param
```

Comment:
Does this need to be seperate and not just passed to the test via pytest parameterize?

Reply:
Also what about the log debugging messages? Can we test those as well?

Reply:
> Do we really need to test the log debugging messages ? . Its too obvious to test it ,isnt it ?

How so?

Reply:
> > Does this need to be seperate and not just passed to the test via pytest parameterize?
> > With a message as well.
> 
> This fixture is behaving same as parametrize . Do we specifcally need to use parametrize ? even if this does the job well ?

A fixture is useful it needs to be used multiple times. If it just one test, parametrize is a better approach, what do you think? Will we use this more than once?

Reply:
> > > Do we really need to test the log debugging messages ? . Its too obvious to test it ,isnt it ?
> > 
> > 
> > How so?
> 
> Wait , Like the code block we want to test is too simple . Its just log and continue. There is almost no chance that this code could go wrong I suppose .

Fair, we will leave that alone then.

Reply:
> > > > Does this need to be seperate and not just passed to the test via pytest parameterize?
> > > > With a message as well.
> > > 
> > > 
> > > This fixture is behaving same as parametrize . Do we specifcally need to use parametrize ? even if this does the job well ?
> > 
> > 
> > A fixture is useful it needs to be used multiple times. If it just one test, parametrize is a better approach, what do you think? Will we use this more than once?
> 
> I dont mind changing it to parametrize , but maybe in future , there might be more than just except raising for the errors , so fixture might be helpful for other tests .

Fair but in that case, we should tidy it up, make it clear what each param is etc

Reply:
Comments and renames

Reply:
Oh yes we did but then I saw the lack of coverage via codecov and changed my mind. We only need to do it for the 404.

Reply:
Thanks